engines/shared/app/errors/shared/bad_request_error.rb

```diff
@@ -0,0 +1,26 @@
+module Shared
+  class BadRequestError
```

Comment:
adds an unnecessary nesting that would require adding it to the autoload conf. Now, we don't need to configure anything since `errors/` it's direct child of `app/`.